virttest/qemu_vm.py

```diff
@@ -4315,6 +4317,24 @@ def _set_migrate_capability(vm, capability, value, is_src_vm=True):
         # "preconfig" is meaningless for dest, remove it whatever the value is
         if "qemu_preconfig" in clone.params:
             del clone.params["qemu_preconfig"]
+        # FIXME: localhost migration should handle all devices params automatically
+        if "tpms" in clone.params:
+            swtpm_dir = os.path.join(data_dir.get_data_dir(), "swtpm")
+            tpms = clone.params.objects("tpms")
+            tpms_copy = []
+            for tpm in tpms:
+                tpm_copied = tpm + "_copied"
+                tpm_path = os.path.join(swtpm_dir, '%s_state' % tpm)
+                tpm_copied_path = os.path.join(swtpm_dir, '%s_state' % tpm_copied)
+                if os.path.exists(tpm_copied_path):
+                    shutil.rmtree(tpm_copied_path)
+                    shutil.copytree(tpm_path, tpm_copied_path)
+                tpms_copy.append(tpm_copied)
+                for param in clone.params.copy():
+                    if param.endswith("_" + tpm):
+                        clone.params[param + "_copied"] = clone.params[param]
+                        clone.params["tpm_overwrite_%s_copied" % tpm] = "no"
```

Comment:
```suggestion
                        tpm_overwrite = clone.params.get("tpm_overwrite_%s_copied" % tpm)
                        if not tpm_overwrite or tpm_overwrite == "yes":
                            clone.params["tpm_overwrite_%s_copied" % tpm] = "no"
```

Reply:
But `tpm_overwrite_%s_copied` is a new parameter for the copied TPM device, so it is always equal to `None`.

Reply:
> But `tpm_overwrite_%s_copied` is a new parameter for the copied TPM device, so it is always equal to `None`.


Yes, I mean that we don't need to assign the value to `"tpm_overwrite_%s_copied" % tpm` repeatedly, only once.
Since you will do it when running `for param in clone.params.copy(): if param.endswith("_" + tpm):`

Reply:
For example:
```
    tpms = tpm0
    tpm_type_tpm0 = emulator
    tpm_model_tpm0 = tpm-crb
    tpm_version_tpm0 = 2.0
```

Reply:
Okay, I got your meaning, you are right, this is a stupid mistake of mine, it should be outside this `for` loop, I will change it.

Reply:
Hi , @PaulYuuu 
The position of suggestion codes is wrong.
The right:
                for param in clone.params.copy():
                    if param.endswith("_" + tpm):
                        clone.params[param + "_copied"] = clone.params[param]
                        tpm_overwrite = clone.params.get("tpm_overwrite_%s_copied" % tpm)
                        if not tpm_overwrite or tpm_overwrite == "yes":
                            clone.params["tpm_overwrite_%s_copied" % tpm] = "no"